KIC 3.0: Introduction (#6356)
Kong/docs.konghq.com: app/_src/kubernetes-ingress-controller/gateway-api.md

purpose: |
  Introduce the Kubernetes Gateway API as our preferred configuration method. Explain Gateway, GatewayClass and provide a simple HTTPRoute example
---

Kong are proud to be a driving force behind the Kubernetes Gateway API standard. With multiple contributors in maintainer and reviewer roles, we're all-in on Gateway API as the future of Kubernetes networking.

{{ site.kic_product_name }} was the first submitted conformance report, and is 100% compliant with the core conformance tests (in addition to many extended tests). Kong has implemented the Gateway API resources as first-class citizens, converting them directly in to {{ site.base_gateway }} configuration rather than using intermediate CRDs. This makes the Gateway API CRDs a native language for {{ site.kic_product_name }}.

The `Ingress` resource will continue to be supported in {{ site.kic_product_name }}, but we highly recommend that new users adopt Gateway API resources such as `HTTPRoute`.

## GatewayClass + Gateway

> The `GatewayClass` object performs the same duties as the `IngressClass resource`. A `GatewayClass` represents a set of Gateways that are managed by the same Ingress Controller. They share any values provided in the `GatewayClass.spec.parametersRef` which control how the Gateways are deployed.

Author: ```suggestion
The `GatewayClass` object performs the same duties as the `IngressClass` resource. A `GatewayClass` represents a set of Gateways that are managed by the same Ingress Controller. They share any values provided in the `GatewayClass.spec.parametersRef` which control how the Gateways are deployed.
```

purpose: |
  Introduce the Kubernetes Gateway API as our preferred configuration method. Explain Gateway, GatewayClass and provide a simple HTTPRoute example
---

> Kong are proud to be a driving force behind the Kubernetes Gateway API standard. With multiple contributors in maintainer and reviewer roles, we're all-in on Gateway API as the future of Kubernetes networking.

Reviewer: This is entirely honest, we are heavily invested in Gateway API, hold (and actively participate) in many key roles, and have been doing so since it's inception @ San Diego Kubecon in 2019.

You could probably even say more.

> Kong are proud to be a driving force behind the Kubernetes Gateway API standard. With multiple contributors in maintainer and reviewer roles, we're all-in on Gateway API as the future of Kubernetes networking.

Reviewer: ```suggestion
We at Kong are proud to be a significant driving force behind the Kubernetes Gateway API standard. Our engineers helped originate the project, and we have been active as maintainers, approvers and reviewer roles from the start. We're all-in on Gateway API as the future of Kubernetes networking.
```

The `Ingress` resource will continue to be supported in {{ site.kic_product_name }}, but we highly recommend that new users adopt Gateway API resources such as `HTTPRoute`.

## GatewayClass + Gateway

The `GatewayClass` object performs the same duties as the `IngressClass resource`. A `GatewayClass` represents a set of Gateways that are managed by the same Ingress Controller. They share any values provided in the `GatewayClass.spec.parametersRef` which control how the Gateways are deployed.

A `Gateway` is a 1:1 mapping to the deployment of a hardware or software load balancer. With {{ site.kic_product_name }} a `Gateway` corresponds to a deployment of {{ site.base_gateway }}.

When using {{ site.kic_product_name }} without the [{{ site.kgo_product_name }}](/gateway-operator/latest/) the `GatewayClass` has a `konghq.com/gatewayclass-unmanaged: 'true'` annotation to indicate that it is manually configured.

> To create a `GatewayClass` and `Gateway` for use with {{ site.kic_product_name }} run the following:

Reviewer 2: ```suggestion
To create a `GatewayClass` and `Gateway` with {{ site.kic_product_name }} run the following:
```

```yaml
echo "
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: GatewayClass
metadata:
  name: kong
  annotations:
    konghq.com/gatewayclass-unmanaged: 'true'

spec:
  controllerName: konghq.com/kic-gateway-controller
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: kong
spec:
  gatewayClassName: kong
  listeners:
  - name: proxy
    port: 80
    protocol: HTTP
" | kubectl apply -f -
```

## Available Route Types

{{ site.kic_product_name }} supports multiple Gateway API route types:

* [`HTTPRoute`](/kubernetes-ingress-controller/{{ page.release }}/guides/services/http/)
* [`TCPRoute`](/kubernetes-ingress-controller/{{ page.release }}/guides/services/tcp/)
* [`UDPRoute`](/kubernetes-ingress-controller/{{ page.release }}/guides/services/udp/)
* [`GRPCRoute`](/kubernetes-ingress-controller/{{ page.release }}/guides/services/grpc/)

> For more information, select a route type above. 

Reviewer 2: ```suggestion

```
